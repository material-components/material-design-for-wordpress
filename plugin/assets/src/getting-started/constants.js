/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const ACTIONS = {
	NEXT_STEP: 'NEXT_STEP',
	GOTO_STEP: 'GOTO_STEP',
	MARK_COMPLETE: 'MARK_COMPLETE',
	ERROR: 'ERROR',
	SET_THEME_OK: 'SET_THEME_OK',
	SET_DEMO_OK: 'SET_DEMO_OK',
	INSTALL_THEME: 'INSTALL_THEME',
	ACTIVATE_THEME: 'ACTIVATE_THEME',
	INSTALL_DEMO_CONTENT: 'INSTALL_DEMO_CONTENT',
	THEME_INSTALLED: 'THEME_INSTALLED',
	DEMO_INSTALLED: 'DEMO_INSTALLED',
};

/**
 * @todo Probably better if we turn these into components like the wizard
 */
export const TABS = {
	WIZARD: __( 'Quick Start', 'material-design' ),
	CUSTOMIZE: __( 'Customize your Theme', 'material-design' ),
	OVERVIEW: __( 'Build with Material Blocks', 'material-design' ),
	UPDATES: __( 'Updates', 'material-design' ),
};
