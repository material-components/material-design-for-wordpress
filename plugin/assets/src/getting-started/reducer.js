/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { ACTIONS, TABS } from './constants';

/**
 * Actions to be taken during the app's life circle
 *
 * @param {*} state  Current state of the world
 * @param {*} action Action dispatched
 */
export const reducer = ( state, action ) => {
	const { type, payload } = action;
	const { activeTab, completed } = state;

	if ( ACTIONS.GOTO_STEP === type ) {
		return { ...state, activeTab: payload.value };
	}

	if ( ACTIONS.MARK_COMPLETE === type ) {
		return { ...state, completed: payload.value };
	}

	if ( ACTIONS.NEXT_STEP === type ) {
		const steps = Object.keys( TABS );
		const stepIndex = steps.indexOf( activeTab );

		if ( stepIndex + 1 === steps.length ) {
			return state;
		}

		return {
			...state,
			activeTab: steps[ stepIndex + 1 ],
			completed: [ activeTab, ...completed ],
			status: STATUS.IDLE,
		};
	}

	if ( ACTIONS.PREVIOUS_STEP === type ) {
		const steps = Object.keys( TABS );
		const stepIndex = steps.indexOf( activeTab );
		let newState = { ...state };
		const newActive = steps[ stepIndex - 1 ];

		newState = {
			...state,
			completed: completed.filter( item => item !== newActive ),
		};

		newState.activeTab = newActive;

		return newState;
	}

	if ( ACTIONS.INSTALL_THEME === type ) {
		return {
			...state,
			status: STATUS.PENDING,
			actionToInstall: ACTIONS.ACTIVATE_THEME,
		};
	}

	if ( ACTIONS.THEME_INSTALLED === type ) {
		return { ...state, themeStatus: 'ok' };
	}

	if ( ACTIONS.DEMO_INSTALLED === type ) {
		return { ...state, contentStatus: 'ok' };
	}

	if ( ACTIONS.ACTIVATE_THEME === type ) {
		return {
			...state,
			actionToInstall: ACTIONS.ACTIVATE_THEME,
			status: STATUS.PENDING,
		};
	}

	if ( ACTIONS.INSTALL_DEMO_CONTENT === type ) {
		return {
			...state,
			actionToInstall: ACTIONS.INSTALL_DEMO_CONTENT,
			status: STATUS.PENDING,
		};
	}

	if ( ACTIONS.SET_THEME_OK === type ) {
		return { ...state, themeStatus: 'ok' };
	}

	if ( ACTIONS.SET_DEMO_OK === type ) {
		return { ...state, contentStatus: 'ok' };
	}

	return state;
};
