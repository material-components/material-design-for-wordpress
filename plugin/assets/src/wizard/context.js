/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { createContext, useReducer } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { reducer } from './reducer';
import { STEPS, STATUS, ADDONS } from './constants';

/**
 * Setups context to be used across the app
 */
const StepContext = createContext();

const { Provider } = StepContext;

/**
 * Default state of the world
 *
 */
const initialState = {
	addons: Object.keys( ADDONS ),
	active: STEPS.WELCOME,
	previous: [],
	status: STATUS.IDLE,
	error: {},
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const StepProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default StepContext;
