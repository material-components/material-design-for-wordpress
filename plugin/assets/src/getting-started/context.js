/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { useReducer, createContext } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { reducer } from './reducer';
import { TABS } from './constants';

/**
 * Setups context to be used across the app
 */
const TabContext = createContext();

const { Provider } = TabContext;

const tabs = Object.keys( TABS );

/**
 * Default state of the world
 *
 */
const initialState = {
	activeTab: 'CUSTOMIZE',
	completed: [ 'WIZARD' ],
	status: STATUS.IDLE,
	actionToInstall: null,
	error: {},
	themeStatus: 'install',
	contentStatus: 'install',
	tabs,
};

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const TabProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default TabContext;
