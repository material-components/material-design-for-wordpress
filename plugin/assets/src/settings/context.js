/**
 * WordPress dependencies
 */
import { useReducer, createContext } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { STATUS } from '../wizard/constants';
import { UPDATERS } from './constants';
import { reducer } from './reducer';
import getConfig from '../admin/get-config';

/**
 * Setups context to be used across the app
 */
const SettingsContext = createContext();

const { Provider } = SettingsContext;

const initialState = {
	status: STATUS.IDLE,
	apiStatus: getConfig( 'apiStatus' ),
	updates: {},
	availableUpdates: Object.keys( UPDATERS ),
	updaters: UPDATERS,
	error: {},
};

initialState.updates[ UPDATERS.FONTS.type ] = true;
initialState.updates[ UPDATERS.ICONS.type ] = false;

/**
 * Creates a wrapper in order to use our custom reducer
 *
 * @param {*} param Children to render
 */
export const SettingsProvider = ( { children } ) => {
	const [ state, dispatch ] = useReducer( reducer, initialState );

	return <Provider value={ { state, dispatch } }>{ children }</Provider>;
};

export default SettingsContext;
