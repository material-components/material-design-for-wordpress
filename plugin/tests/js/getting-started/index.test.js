/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

/**
 * Internal dependencies
 */
import GettingStarted from '../../../assets/src/getting-started/components/index';
import { TabProvider } from '../../../assets/src/getting-started/context';

const setup = () => {
	return render(
		<TabProvider>
			<GettingStarted />
		</TabProvider>
	);
};

describe( 'GSM: Getting Started', () => {
	beforeAll( () => {
		global.materialDesignGsm = {
			themeStatus: 'ok',
			contentStatus: 'ok',
		};
	} );

	afterEach( cleanup );

	it( 'matches snapshot', () => {
		const wrapper = setup();
		expect( wrapper ).toMatchSnapshot();
	} );
} );
