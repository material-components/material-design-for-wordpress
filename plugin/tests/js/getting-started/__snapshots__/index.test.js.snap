// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSM: Getting Started matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-layout-grid mdc-typography"
      >
        <div
          class="mdc-layout-grid__inner"
        >
          <div
            class="material-gsm__navigation mdc-layout-grid__cell mdc-layout-grid__cell--span-3"
          >
            <div
              class="material-gsm__heading"
            >
              <div
                class="mdc-typography--headline6"
              >
                Getting Started
              </div>
            </div>
            <div
              class="material-gsm__tabs"
            >
              <button
                class="material-gsm__tab material-gsm__tab-completed"
                type="button"
              >
                <i
                  class="material-icons"
                >
                  check_circle_outline
                </i>
                <div
                  class="material-gsm__tab-title"
                >
                  Quick Start
                </div>
              </button>
              <button
                class="material-gsm__tab material-gsm__tab-active"
                type="button"
              >
                <i
                  class="material-icons"
                >
                  lens
                </i>
                <div
                  class="material-gsm__tab-title"
                >
                  Customize your Theme
                </div>
              </button>
              <button
                class="material-gsm__tab"
                type="button"
              >
                <i
                  class="material-icons"
                >
                  lens
                </i>
                <div
                  class="material-gsm__tab-title"
                >
                  Build with Material Blocks
                </div>
              </button>
              <button
                class="material-gsm__tab"
                type="button"
              >
                <i
                  class="material-icons"
                >
                  lens
                </i>
                <div
                  class="material-gsm__tab-title"
                >
                  Updates
                </div>
              </button>
            </div>
          </div>
          <div
            class="material-gsm__content mdc-layout-grid__cell mdc-layout-grid__cell--span-9"
          >
            <h2
              class="material-gsm__content-title mdc-typography--headline3"
            >
              Customize your Theme
            </h2>
            <p>
              Set and preview your global theme styles using the Material Design options in the Customize panel. Choose colors, typography, shapes, and icons to express your unique style. Once you’re satisfied with your changes, hit “Publish” to update your site.
            </p>
            <img
              alt=""
              src="undefinedcustomize-your-theme.png"
              style="max-width: 40%;"
            />
            <div
              style="height: 15px;"
            />
            <div
              class="material-gsm__content-actions"
            >
              <button
                class="mdc-button mdc-button--raised"
              >
                <span
                  class="mdc-button__label"
                >
                  Customize
                </span>
                <i
                  aria-hidden="true"
                  class="material-icons mdc-button__icon trailing-icon"
                >
                  navigate_next
                </i>
              </button>
            </div>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Starter Styles
            </h2>
            <p>
              Start from our existing styles and use 
              <a
                href="https://material.io/design/material-theming/overview.html"
                rel="noopener noreferrer"
                target="_blank"
              >
                Material Theming
              </a>
               to create a custom look and feel.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Color Palettes
            </h2>
            <p>
              Change your primary and secondary colors and see them applied throughout your site. Need help picking colors? Try Material’s 
              <a
                href="https://material.io/design/color/the-color-system.html#tools-for-picking-colors"
                rel="noopener noreferrer"
                target="_blank"
              >
                color palette generator
              </a>
               to find complementary, analogous, and triadic tonal palettes.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Typography
            </h2>
            <p>
              Choose from more than 1,000 typefaces in 
              <a
                href="https://fonts.google.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Google Fonts
              </a>
               to set your headline and body styles. Want tips on balancing expression with legibility?. Read Material’s 
              <a
                href="https://material.io/design/typography/the-type-system.html"
                rel="noopener noreferrer"
                target="_blank"
              >
                typography guidelines
              </a>
               for inspiration and best practices.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Corner Styles
            </h2>
            <p>
              Material Design systematically applies a shape scheme across components. Change the global corner radius for all blocks or individually for each Material block.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Icon Style
            </h2>
            <p>
              All 
              <a
                href="http://material.io/icons"
                rel="noopener noreferrer"
                target="_blank"
              >
                Material Icons
              </a>
               are included with this plugin. Choose from Filled, Outlined, Rounded, Two-tone, or Sharp styles.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline3"
            >
              Set Layout & Navigation
            </h2>
            <p>
              Select how posts display – as either cards or image lists, in a wide or normal width. Additional card display options include raised or outlined cards, and show or hide post comments, author, excerpt, and date. 
              Choose how comment fields display within each post.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Top app bar
            </h2>
            <p>
              Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site from within the top app bar. By default, the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar visible at all times.
            </p>
            <h2
              class="material-gsm__content-title mdc-typography--headline5"
            >
              Footer
            </h2>
            <p>
              Add footer text and give your users a quick way to jump back to the top of the page. Looking to add widgets at the bottom of your site? Use the 
              <a
                href="\${ getConfig( 'customize' ) }?autofocus[panel]=widgets"
                rel="noopener noreferrer"
                target="_blank"
              >
                Widgets
              </a>
               settings.
            </p>
            <div
              style="height: 20px;"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-layout-grid mdc-typography"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="material-gsm__navigation mdc-layout-grid__cell mdc-layout-grid__cell--span-3"
        >
          <div
            class="material-gsm__heading"
          >
            <div
              class="mdc-typography--headline6"
            >
              Getting Started
            </div>
          </div>
          <div
            class="material-gsm__tabs"
          >
            <button
              class="material-gsm__tab material-gsm__tab-completed"
              type="button"
            >
              <i
                class="material-icons"
              >
                check_circle_outline
              </i>
              <div
                class="material-gsm__tab-title"
              >
                Quick Start
              </div>
            </button>
            <button
              class="material-gsm__tab material-gsm__tab-active"
              type="button"
            >
              <i
                class="material-icons"
              >
                lens
              </i>
              <div
                class="material-gsm__tab-title"
              >
                Customize your Theme
              </div>
            </button>
            <button
              class="material-gsm__tab"
              type="button"
            >
              <i
                class="material-icons"
              >
                lens
              </i>
              <div
                class="material-gsm__tab-title"
              >
                Build with Material Blocks
              </div>
            </button>
            <button
              class="material-gsm__tab"
              type="button"
            >
              <i
                class="material-icons"
              >
                lens
              </i>
              <div
                class="material-gsm__tab-title"
              >
                Updates
              </div>
            </button>
          </div>
        </div>
        <div
          class="material-gsm__content mdc-layout-grid__cell mdc-layout-grid__cell--span-9"
        >
          <h2
            class="material-gsm__content-title mdc-typography--headline3"
          >
            Customize your Theme
          </h2>
          <p>
            Set and preview your global theme styles using the Material Design options in the Customize panel. Choose colors, typography, shapes, and icons to express your unique style. Once you’re satisfied with your changes, hit “Publish” to update your site.
          </p>
          <img
            alt=""
            src="undefinedcustomize-your-theme.png"
            style="max-width: 40%;"
          />
          <div
            style="height: 15px;"
          />
          <div
            class="material-gsm__content-actions"
          >
            <button
              class="mdc-button mdc-button--raised"
            >
              <span
                class="mdc-button__label"
              >
                Customize
              </span>
              <i
                aria-hidden="true"
                class="material-icons mdc-button__icon trailing-icon"
              >
                navigate_next
              </i>
            </button>
          </div>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Starter Styles
          </h2>
          <p>
            Start from our existing styles and use 
            <a
              href="https://material.io/design/material-theming/overview.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              Material Theming
            </a>
             to create a custom look and feel.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Color Palettes
          </h2>
          <p>
            Change your primary and secondary colors and see them applied throughout your site. Need help picking colors? Try Material’s 
            <a
              href="https://material.io/design/color/the-color-system.html#tools-for-picking-colors"
              rel="noopener noreferrer"
              target="_blank"
            >
              color palette generator
            </a>
             to find complementary, analogous, and triadic tonal palettes.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Typography
          </h2>
          <p>
            Choose from more than 1,000 typefaces in 
            <a
              href="https://fonts.google.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Google Fonts
            </a>
             to set your headline and body styles. Want tips on balancing expression with legibility?. Read Material’s 
            <a
              href="https://material.io/design/typography/the-type-system.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              typography guidelines
            </a>
             for inspiration and best practices.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Corner Styles
          </h2>
          <p>
            Material Design systematically applies a shape scheme across components. Change the global corner radius for all blocks or individually for each Material block.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Icon Style
          </h2>
          <p>
            All 
            <a
              href="http://material.io/icons"
              rel="noopener noreferrer"
              target="_blank"
            >
              Material Icons
            </a>
             are included with this plugin. Choose from Filled, Outlined, Rounded, Two-tone, or Sharp styles.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline3"
          >
            Set Layout & Navigation
          </h2>
          <p>
            Select how posts display – as either cards or image lists, in a wide or normal width. Additional card display options include raised or outlined cards, and show or hide post comments, author, excerpt, and date. 
            Choose how comment fields display within each post.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Top app bar
          </h2>
          <p>
            Choose tabs, navigation drawer, or both for your site navigation. Give users the ability to search your site from within the top app bar. By default, the top app bar will hide when scrolling up, choose “Fixed” to keep the app bar visible at all times.
          </p>
          <h2
            class="material-gsm__content-title mdc-typography--headline5"
          >
            Footer
          </h2>
          <p>
            Add footer text and give your users a quick way to jump back to the top of the page. Looking to add widgets at the bottom of your site? Use the 
            <a
              href="\${ getConfig( 'customize' ) }?autofocus[panel]=widgets"
              rel="noopener noreferrer"
              target="_blank"
            >
              Widgets
            </a>
             settings.
          </p>
          <div
            style="height: 20px;"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
